Recognize hook on record beep

Handballermann opened the pull request:
After building the audio-guestbook and using it on a wedding for the first time, I came across an issue: If you put the hook back, while the _beep_ before the recording is played, the hook press is not recognized. The problem is: while in the `wait()`-function, button presses are recognized, but not communicated to outside of `wait()`.
I found some issues for this problem, but was not convinced of the solutions (max recording time 120seconds), so I made this one.

I made `wait()` return `boolean` instead of `void`. If the `recordButton` is pressed, the hook is back on it (raising/falling edge still need to be checked for your specific telephone). If the hook is back on, the `wait()`-function is interrupted and returns `false`. If `wait()` was able to run through, it returns `true`.
Before starting the record, it's checked if `wait()` is running through without interrupts (returning `true`). If so, start recording. If `wait()` is interrupted and returns `false`, immediately stop the _beep_ and get into mode `Ready`.

Handballermann committed:
let wait return interrupts by hook

recordButton press means, the hook is back on it (make sure to have your raising/falling edge right).
If the hook is back on, the wait function is interrupted and returns false. If wait was able to run through, it returns true.

Handballermann committed:
add comment with returns to wait

Handballermann committed:
use boolean returns from wait

before starting the record, check if while is running through without interrupts (returning true). If so, start recording. If wait is interrupted and returns false, immediately stop the beep and get into ready mode

h4yn0nnym0u5e commented:
Also fixed in PR #47 back in May. No point, really: nothing's been pulled since last November... it's not even as if the account is inactive, either.

Handballermann committed:
add wait to taking phone of the hook

if the hook is taken of and immediately returned, the greeting will no more be played.

Handballermann committed:
recognize hook hanging up on beep while playing all messages

untested

Handballermann commented:
@h4yn0nnym0u5e I updated my branch
I also gonna update my personal phones with some code of your PR :)
